src/components/CodeInput.tsx

```diff
@@ -61,6 +61,12 @@ export default function CodeInput({
 }: Props) {
   const [forceShowingPasteIcon, clipboardContent, getFreshClipboardContent] = useClipboard()
 
+  const clipboardRef = useRef(clipboardContent)
```

Comment:
i need to rework this part, the ref resolves the main issue but does lag one render cycle behind so the paste button doesn't appear right away :/ any ideas appreciated